components/merkledb/src/proof_list_index/mod.rs

```diff
@@ -505,6 +488,49 @@ where
         }
     }
 
+    /// Updates levels of the tree with heights `2..` after the values in the range
+    /// `[first_index, last_index]` were updated.
+    ///
+    /// # Invariants
+    ///
+    /// - `self.len()` / `self.height()` is assumed to be correctly set.
+    /// - Value hashes (i.e., tree branches on level 1) are assumed to be updated.
+    fn update_range(&mut self, mut first_index: u64, mut last_index: u64) {
+        let mut last_level_index = self.len() - 1;
```

Comment:
I'd name it `last_index_of_level` to avoid ambiguity (now it can be considered to be an index of a last level).

Reply:
I'd suggest even something like `last_element_index`. It's the same thing, isn't it? (I'm literally blowing my brain now)

Reply:
`last_element_index` may be considered to be the index of the last element in the list, which is only true for the first iteration of the loop below.